entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/identifier/IdentifierConverterFactoryImpl.java

```diff
@@ -63,18 +65,24 @@ public IdentifierConverter getIdentifierConverter(
 
   private boolean isFilteringFieldArray(
       final RequestContext requestContext, final String columnId) {
-    return attributeMapping.isMultiValued(requestContext, columnId);
+    final Optional<AttributeKind> attributeKind =
+        attributeMapping.getAttributeKind(requestContext, columnId);
+    return attributeKind.isPresent() && attributeMapping.isArray(attributeKind.get());
   }
 
   private boolean isFilterValueArray(final ValueType valueType) {
     return valueHelper.isArray(valueType);
   }
 
-  private boolean isFilterValueMap(final ValueType valueType) {
-    return valueHelper.isMap(valueType);
+  private boolean isFilteringFieldMap(final RequestContext context, final String columnId) {
+    final Optional<AttributeKind> attributeKind =
+        attributeMapping.getAttributeKind(context, columnId);
+    return attributeKind.isPresent() && attributeMapping.isMap(attributeKind.get());
   }
 
-  private boolean isFilterValuePrimitive(final ValueType valueType) {
-    return valueHelper.isPrimitive(valueType);
+  private boolean isFilteringFieldPrimitive(final RequestContext context, final String columnId) {
+    final Optional<AttributeKind> attributeKind =
+        attributeMapping.getAttributeKind(context, columnId);
+    return attributeKind.isPresent() && attributeMapping.isPrimitive(attributeKind.get());
```

Comment:
Tip: Can be simplified
```suggestion
    return attributeKind.map(attributeMapping::isPrimitive).orElse(false);
```

Reply:
Updated accordingly in all the `if()` conditions as well.